bokehjs/src/lib/core/util/arrayable.ts

```diff
@@ -35,7 +38,20 @@ export function copy<T>(array: Arrayable<T>): Arrayable<T> {
   }
 }
 
+export function splice<T>(array: T[], start: number, k?: number, ...items: T[]): T[]
+export function splice<T>(array: Arrayable<T>, start: number, k?: number, ...items: T[]): Arrayable<T>
+
 export function splice<T>(array: Arrayable<T>, start: number, k?: number, ...items: T[]): Arrayable<T> {
+  if (Array.isArray(array)) {
+    const result = copy(array)
+    if (k === undefined) {
+      result.splice(start)
+    } else {
+      result.splice(start, k, ...items)
```

Comment:
Are there any reasons not to use toSpliced here?

Reply:
`toSpliced` is a very new API. It needs to be assessed if we can use it already.

Reply:
It's so new in fact, that TypeScirpt's standard library doesn't have type definitions for it. Not that it's a problem, because we can always extend them with our own.

Reply:
Looks like it was added in the last two years. My gut feeling is that it is too early to add it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced#browser_compatibility